### PR TITLE
Fix gtkplus build by hardcoding X11 support

### LINKED ISSUE #6940: gtkplus build failure on Ubuntu LTS
On a clean `Ubuntu 16.04.3 LTS` (built via [this Dockerfile](https://gist.github.com/nazavode/50c6336f73fca09cf0c4b0997747df08)) the `configure` phase of `gtkplus` fails complaining about missing `cairo-xlib`. When `gtkplus` is asked for `+X` variant, everything works as expected as `cairo-xlib` is provided by `cairo+X`. I tried to configure the source manually using `./configure --without-x` but it keeps complaining: `cairo-xlib` looks like a mandatory dependency.

### Expected Result

To have the `gtkplus` package built and installed.

### Actual Result

```
==> Error: ProcessError: Command exited with status 1:
    '/home/spack/spack/var/spack/stage/gtkplus-2.24.31-hbqdviwkldktib7stpuem5mapww527hb/gtk+-2.24.31/configure' '--prefix=/home/spack/spack/opt/spack/linux-ubuntu16.04-x86_64/gcc-5.4.0/gtkplus-2.24.31-hbqdviwkldktib7stpuem5mapww527hb'

1 error found in build log:
     [ ... ]
     106   checking for library containing strerror... none required
     107   checking whether make sets $(MAKE)... (cached) yes
     108   checking for pkg-config... /home/spack/spack/opt/spack/linux-ubuntu16.04-x86_64/gcc-5.4.0/pkgconf-1.4.0-c5lf2hilcinnqpf3nojkif3eauay2gko/bin/pkg-config
     109   checking pkg-config is at least version 0.9.0... yes
     110   checking for BASE_DEPENDENCIES... yes
     111   checking for CAIRO_BACKEND... no
  >> 112   configure: error: Package requirements (cairo-xlib >= 1.6) were not met:
     113
     114   Package 'cairo-xlib', required by 'virtual:world', not found
     115
     116   Consider adjusting the PKG_CONFIG_PATH environment variable if you
     117   installed software in a non-standard prefix.
     118
```

### Steps to reproduce the issue

```console
$ spack install gtkplus
```

### Information on your system

The system used to reproduce this issue has been built via [this Dockerfile](https://gist.github.com/nazavode/50c6336f73fca09cf0c4b0997747df08). It is an `Ubuntu 16.04.3 LTS Server` (no GUI components) running `gcc@5.4.0` and `python@2.7.12`.

### PR DESCRIPTION
**Pull request by nazavode**
Fixes #6940 by hardconding X11 support as graphical backend.

### CODE REVIEW

**Comment by alalazo** [comment]
@lee218llnl @adamjstewart Requested your review. If you are fine with the changes, I'll merge this.

**Comment by adamjstewart** [comment]
I'm back in school now so I won't have much time to review PRs anytime soon.

**Comment by lee218llnl** [comment]
There are a few packages that depend_on('gtkplus+X') that should probably be updated along with this PMR (i.e., remove the X variant).

bash-4.1$ grep gtkplus -nr var/spack/repos/builtin/packages/*/*.py | grep +X
var/spack/repos/builtin/packages/dia/package.py:36:    depends_on('gtkplus@2.6.0:+X')
var/spack/repos/builtin/packages/emacs/package.py:60:    depends_on('gtkplus+X', when='+X toolkit=gtk')
var/spack/repos/builtin/packages/icedtea/package.py:61:    depends_on('gtkplus+X', when='~X', type='build')
var/spack/repos/builtin/packages/icedtea/package.py:74:    depends_on('gtkplus+X', when='+X')
var/spack/repos/builtin/packages/py-pygtk/package.py:41:    depends_on('gtkplus+X@2.24:2.99')
var/spack/repos/builtin/packages/slurm/package.py:71:    depends_on('gtkplus+X', when='+gtk')
var/spack/repos/builtin/packages/wx/package.py:50:    depends_on('gtkplus+X')

**Comment by nazavode** [comment]
@lee218llnl you are right! I totally forgot to fix all dependents along `gtkplus`, my bad, thanks. @alalazo @lee218llnl could you pls mark as `don't-merge-yet`/`wip` until I've tested all the modified packages? Thanks.

File: var/spack/repos/builtin/packages/icedtea/package.py [hunk 1]
```diff
@@ -58,7 +58,7 @@ class Icedtea(AutotoolsPackage):
     depends_on('libxcomposite', when='~X', type='build')
     depends_on('libxau', when='~X', type='build')
     depends_on('libxdmcp', when='~X', type='build')
-    depends_on('gtkplus+X', when='~X', type='build')
```
Comment (alalazo) [inline var/spack/repos/builtin/packages/icedtea/package.py]: I assume that, incidentally, we fixed a bug :+1:

Comment (nazavode) [reply var/spack/repos/builtin/packages/icedtea/package.py]: I think that the `+X` deps are needed anyway, they all become `build` only when `~X` is requested on `icedtea`:

```
# X11 deps required for building even when headless
```

**Comment by nazavode** [comment]
@lee218llnl @alalazo tested the affected packages on `arch=linux-ubuntu16.04-x86_64`, the changes seems to be working correctly. Please note that in order to have `slurm` built, #7048 (unrelated to this PR) must be applied as well.

**Comment by alalazo** [comment]
@lee218llnl If it is fine with you, I'll merge the PR

**Comment by adamjstewart** [comment]
This seems to have broken other important packages:
```console
$ spack spec r
...
gtkplus requires pango variant +X, but spec asked for ~X
```

**Comment by alalazo** [comment]
@adamjstewart Can you check if there's some custom change you made to `packages.yaml`? What I get on my laptop seems correct:
```console
$ spack spec r
Input spec
--------------------------------
r

Concretized
--------------------------------
r@3.5.0%gcc@8~X~external-lapack arch=linux-ubuntu18.04-x86_64 
    ^bzip2@1.0.6%gcc@8+shared arch=linux-ubuntu18.04-x86_64 
    ^cairo@1.14.12%gcc@8~X arch=linux-ubuntu18.04-x86_64 
        ^fontconfig@2.12.3%gcc@8 arch=linux-ubuntu18.04-x86_64 
            ^font-util@1.3.1%gcc@8 arch=linux-ubuntu18.04-x86_64 
                ^pkgconf@1.4.2%gcc@8 arch=linux-ubuntu18.04-x86_64 
                ^util-macros@1.19.1%gcc@8 arch=linux-ubuntu18.04-x86_64 
            ^freetype@2.7.1%gcc@8 arch=linux-ubuntu18.04-x86_64 
                ^libpng@1.6.34%gcc@8 arch=linux-ubuntu18.04-x86_64 
                    ^zlib@1.2.11%gcc@8+optimize+pic+shared arch=linux-ubuntu18.04-x86_64 
            ^gperf@3.0.4%gcc@8 arch=linux-ubuntu18.04-x86_64 
            ^libxml2@2.9.8%gcc@8~python arch=linux-ubuntu18.04-x86_64 
                ^xz@5.2.4%gcc@8 arch=linux-ubuntu18.04-x86_64 
        ^glib@2.56.1%gcc@8~libmount patches=c325997b72a205ad1638bb3e3ba0e5b73e3d32ce63b2d0d3282f3e3a2ff4663c tracing= arch=linux-ubuntu18.04-x86_64 
            ^gettext@0.19.8.1%gcc@8+bzip2+curses+git~libunistring+libxml2 patches=9acdb4e73f67c241b5ef32505c9ddf7cf6884ca8ea661692f21dca28483b04b8 +tar+xz arch=linux-ubuntu18.04-x86_64 
                ^ncurses@6.1%gcc@8~symlinks~termlib arch=linux-ubuntu18.04-x86_64 
                ^tar@1.30%gcc@8 arch=linux-ubuntu18.04-x86_64 
            ^libffi@3.2.1%gcc@8 arch=linux-ubuntu18.04-x86_64 
            ^pcre@8.42%gcc@8~jit+utf arch=linux-ubuntu18.04-x86_64 
            ^perl@5.26.2%gcc@8+cpanm patches=0eac10ed90aeb0459ad8851f88081d439a4e41978e586ec743069e8b059370ac +shared+threads arch=linux-ubuntu18.04-x86_64 
                ^gdbm@1.14.1%gcc@8 arch=linux-ubuntu18.04-x86_64 
                    ^readline@7.0%gcc@8 arch=linux-ubuntu18.04-x86_64 
            ^python@2.7.15%gcc@8+dbm~optimizations patches=123082ab3483ded78e86d7c809e98a804b3465b4683c96bd79a2fd799f572244 +pic+pythoncmd+shared~tk~ucs4 arch=linux-ubuntu18.04-x86_64 
                ^openssl@1.0.2o%gcc@8+systemcerts arch=linux-ubuntu18.04-x86_64 
                ^sqlite@3.23.1%gcc@8~functions arch=linux-ubuntu18.04-x86_64 
        ^pixman@0.34.0%gcc@8 arch=linux-ubuntu18.04-x86_64 
    ^curl@7.60.0%gcc@8~darwinssl~libssh~libssh2~nghttp2 arch=linux-ubuntu18.04-x86_64 
    ^icu4c@60.1%gcc@8 arch=linux-ubuntu18.04-x86_64 
    ^jdk@8u172-b11%gcc@8 arch=linux-ubuntu18.04-x86_64 
    ^libjpeg-turbo@1.5.90%gcc@8 arch=linux-ubuntu18.04-x86_64 
        ^cmake@3.11.4%gcc@8~doc+ncurses+openssl+ownlibs~qt arch=linux-ubuntu18.04-x86_64 
        ^nasm@2.13.03%gcc@8 patches=ac9f315d204afa6b99ceefa1fe46d4eed2b8a23c7315d32d33c0f378d930e950 arch=linux-ubuntu18.04-x86_64 
    ^libtiff@4.0.9%gcc@8 arch=linux-ubuntu18.04-x86_64 
    ^pango@1.41.0%gcc@8~X arch=linux-ubuntu18.04-x86_64 
        ^gobject-introspection@1.49.2%gcc@8 patches=6f90bb267efa043ed70b900b4b8e2faf9e8133afae311893b01060356ea81bba arch=linux-ubuntu18.04-x86_64 
            ^bison@3.0.4%gcc@8 patches=b72914fe38e54a6fc25f29019e0a0786705c4f61ce20d414cc2010c8d62448c7 arch=linux-ubuntu18.04-x86_64 
                ^m4@1.4.18%gcc@8 patches=3877ab548f88597ab2327a2230ee048d2d07ace1062efe81fc92e91b7f39cd00,c0a408fbffb7255fcc75e26bd8edab116fc81d216bfd18b473668b7739a4158e +sigsegv arch=linux-ubuntu18.04-x86_64 
                    ^libsigsegv@2.11%gcc@8 arch=linux-ubuntu18.04-x86_64 
            ^flex@2.6.3%gcc@8+lex arch=linux-ubuntu18.04-x86_64 
                ^help2man@1.47.4%gcc@8 arch=linux-ubuntu18.04-x86_64 
        ^harfbuzz@1.4.6%gcc@8 arch=linux-ubuntu18.04-x86_64 
    ^tcl@8.6.8%gcc@8 arch=linux-ubuntu18.04-x86_64 
    ^tk@8.6.8%gcc@8 arch=linux-ubuntu18.04-x86_64 
        ^libx11@1.6.5%gcc@8 arch=linux-ubuntu18.04-x86_64 
            ^inputproto@2.3.2%gcc@8 arch=linux-ubuntu18.04-x86_64 
            ^kbproto@1.0.7%gcc@8 arch=linux-ubuntu18.04-x86_64 
            ^libxcb@1.13%gcc@8 arch=linux-ubuntu18.04-x86_64 
                ^libpthread-stubs@0.4%gcc@8 arch=linux-ubuntu18.04-x86_64 
                ^libxau@1.0.8%gcc@8 arch=linux-ubuntu18.04-x86_64 
                    ^xproto@7.0.31%gcc@8 arch=linux-ubuntu18.04-x86_64 
                ^libxdmcp@1.1.2%gcc@8 arch=linux-ubuntu18.04-x86_64 
                ^xcb-proto@1.13%gcc@8 arch=linux-ubuntu18.04-x86_64 
            ^xextproto@7.3.0%gcc@8 arch=linux-ubuntu18.04-x86_64 
            ^xtrans@1.3.5%gcc@8 arch=linux-ubuntu18.04-x86_64 
```

**Comment by adamjstewart** [comment]
@alalazo Good call. It looks like having JDK as a non-buildable external package causes concretization to fail. I can't for the life of me figure out why.

**Pull request merged**